gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -455,6 +470,54 @@ public synchronized void setActive(boolean active) {
     }
   }
 
+  /**
+   * Used by the DagManager to launch a new execution for a flow action event loaded from the DagActionStore upon
+   * setting this instance of the DagManager to active. Because it may be a completely new DAG not contained in the
+   * dagStore, we compile the flow to generate the dag before calling addDag(), handling any errors that may result in
+   * the process.
+   */
+  public void handleLaunchFlowEvent(DagActionStore.DagAction action) {
+    if (this.specCompiler.isPresent()) {
+      FlowId flowId = new FlowId().setFlowGroup(action.getFlowGroup()).setFlowName(action.getFlowName());
```

Comment:
Same comment here as the one below, wanted to double check that all the checks are being done before submitting these flows which the orchestrator is doing, otherwise there will be some bugs introduced that are guarded against. Is it possible to use a helper or a static function?

Reply:
Re-used helper function created in Orchestrator